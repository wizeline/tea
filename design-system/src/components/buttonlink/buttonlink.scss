wz-button-link {
  display: inline-block;

  &:focus {
    outline: 0;
  }

  button {
    background-color: transparent;
    padding: 0 16px;
    border: none;
    color: var(--primary-base);
    cursor: pointer;
    font-size: 14px;
    cursor: pointer;
    outline: inherit;

    &:hover {
      color: var(--primary-dark);
    }

    &:disabled {
      cursor: default;
      color: var(--ink-light);
    }

 }

}
