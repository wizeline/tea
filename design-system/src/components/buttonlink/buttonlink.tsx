import { Component, Prop, Element } from '@stencil/core';

@Component({
  tag: 'wz-button-link',
  styleUrl: 'buttonlink.scss',
  shadow: false,
})
export class ButtonLink {
  @Element() button: HTMLButtonElement;

  // State props
  @Prop() disabled: boolean;

  render() {

    return (
      <button
        disabled={this.disabled}
        >
        <slot />
      </button>
    );
  }
}
