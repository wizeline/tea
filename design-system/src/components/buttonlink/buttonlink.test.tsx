import { ButtonLink } from './buttonlink';

import { TestWindow } from '@stencil/core/testing';

describe('wz-button-link', () => {
  it('renders value', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [ButtonLink],
      html: '<wz-button-link>Hello, my name is Marty McFly</wz-button-link>',
    });
    const buttonLink = element.querySelector('button');

    expect(buttonLink).toMatchSnapshot();
    expect(buttonLink.innerHTML).toEqual('Hello, my name is Marty McFly');
  });

  it('renders disabled', async () => {
    const window = new TestWindow();
    const element = await window.load({
      components: [ButtonLink],
      html: '<wz-button-link disabled>Hello, my name is Marty McFly</wz-button-link>',
    });
    const buttonLink = element.querySelector('button');

    expect(buttonLink).toMatchSnapshot();
    expect(buttonLink.innerHTML).toEqual('Hello, my name is Marty McFly');
  });
});
