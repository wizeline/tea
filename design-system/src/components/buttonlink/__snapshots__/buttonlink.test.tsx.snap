// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wz-button-link renders disabled 1`] = `
<button
  disabled=""
>
  
  Hello, my name is Marty McFly
</button>
`;

exports[`wz-button-link renders value 1`] = `
<button>
  
  Hello, my name is Marty McFly
</button>
`;
